core/shared/src/main/scala/cats/parse/Parser.scala

```diff
@@ -862,6 +862,32 @@ object Parser extends ParserInstances {
     }
   }
 
+  /** If this right parser fails to parse its input with an epsilon error,
+    * try the left parser instead.
+    *
+    * If the right parser fails with an arresting error, the left parser
+    * won't be tried.
+    *
+    * Backtracking may be used on the left parser to allow the right
+    * one to pick up after any error, resetting any state that was
+    * modified by the left parser.
+    */
+  def either[A, B](left: Parser[A], right: Parser[B]): Parser[Either[A, B]] =
```

Comment:
There was interest in having these. Here I gave preference to the right parser as it's seen as the "success". However, this gave me some doubts regarding the parameter order.

Also, should we just prefer these over the infix ones or keep both?

I believe the infix ones are more explicit in regards to which parser is attempted first and thus the "right" side.

wdyt?

Reply:
1. There are two ideas in tension here: 1 left to right reading order suggests left comes first, 2 convention of Either is that biases to Right.
2. What name should we have, if any, for a method.

In the method case, the "this" has to be the first one read. I tend to think the same for def either. So I guess I would break the normal Right biased-ness. I don't know.

For the method name, I'm pretty concerned about confusion with or and orElse. I still like my suggestion of eitherOr as a method.

Reply:
Let's go with `eitherOr`, revert back to not be right biased and see how that looks.

Reply:
I have to say that `eitherOr` grew on me.

Reply:
@johnynek Add the 2 commits one with and without right bias. I think the latter looks good now.